File: Frontend/src/interfaces/v12/stockAdjustment/stockAdjustmentDetails/stockAdjustmentDetailsDto.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";

export interface StockAdjustmentDetailsDto extends CreatedChangedEntity {
  id: guid;
  number: string;
  warehouseId: guid;
  stockAdjustmentItems: StockAdjustmentItemDetailsDto[];
}
export interface StockAdjustmentItemDetailsDto extends CreatedChangedEntity {
  id: guid;
  stockAdjustmentId: guid;
  productId: guid;
  locationId: guid;
  quantity: number;
  reason: string;
}

File: Wms.Api/Dto/StockAdjustment/StockAdjustmentDetails/StockAdjustmentDetailsDto.cs
using Wms.Api.Dto.StockAdjustment.StockAdjustmentDetails;

namespace Wms.Api.Dto.StockAdjustment.StockAdjustmentCreateUpdate
{
    public class StockAdjustmentDetailsDto
    {
        public Guid Id { get; set; }
        public required string Number { get; set; } 
        public Guid WarehouseId { get; set; }
        public DateTime CreatedAt { get; set; }
        public required string CreatedById { get; set; }
        public DateTime? ChangedAt { get; set; }
        public string? ChangedById { get; set; }
        public ICollection<StockAdjustmentItemDetailsDto>? StockAdjustmentItems { get; set; }
    }
}
